Lib/lib2to3/tests/test_parser.py

```diff
@@ -622,6 +622,13 @@ def test_multiline_str_literals(self):
         self.validate(s)
 
 
+class TestPickleableException(unittest.TestCase):
+    def test_ParseError(self):
+        err = ParseError('msg', 2, None, (1, 'context'))
+        err2 = pickle.loads(pickle.dumps(err))
+        self.assertEqual(vars(err), vars(err2))
```

Comment:
It is better to test attributes explicitly. If use `__slots__`, `vars()` can not work.

Reply:
done!  though in this case this is not a slots object